clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala

```diff
@@ -49,14 +50,12 @@ object UncommittedGarbageCollector {
       .select("address", "last_modified")
 
     // Read objects from namespace root, for old structured repositories
-    val oldDataPath = new Path(storageNamespace)
+
     // TODO (niro): implement parallel lister for old repositories (https://github.com/treeverse/lakeFS/issues/4620)
     val oldDataDF = new NaiveDataLister().listData(configMapper, oldDataPath)
-    dataDF.union(oldDataDF)
+    dataDF =
+      dataDF.union(oldDataDF).filter(dataDF("last_modified") < before.getTime).select("address")
```

Comment:
Will the filtering work correctly? 
The `oldDataDF` isn't part of `dataDF` when the args are captured. I think the `oldDataDF` won't get filtered.

Reply:
This is just to provide the column name - modified to use `col` instead